app/forms/journeys/additional_payments_for_teaching/eligible_itt_subject_form.rb

```diff
@@ -27,15 +27,33 @@ def chemistry_or_physics_available?
 
       def save
         return false unless valid?
-
-        claim.assign_attributes(
-          eligibility_attributes: {eligible_itt_subject: eligible_itt_subject}
-        )
-        claim.reset_eligibility_dependent_answers(["eligible_itt_subject"])
-        claim.save!
+        return true unless eligible_itt_subject_changed?
+
+        if claim.qualifications_details_check?
+          update!(
+            eligibility_attributes: {
+              eligible_itt_subject: eligible_itt_subject
+            }
+          )
+        else
+          # FIXME RL: decide how to handle resetting `eligible_degree_subject`
+          # and add test for it - only LUP eligibility has this field
```

Comment:
Not sure how best to tackle this.
Options are
1.) Update current_claim to filter attributes from eligibility_attributes if the eligibility it's setting attributes for doesn't respond to the method
2.) Add a non persisted attribute to `EarlyCareerPayments::Eligibility` for compatibility
3.) Something else

Maybe `eligible_degree_subject` is a sign that the eligibilities can't be treated the same and we shouldn't try and move reseting dependent answers into the form object?

Any suggestions on this?